src/devices/PiJuice/PiJuiceStatus.cs

```diff
@@ -0,0 +1,262 @@
+
+            int i = BinaryPrimitives.ReadInt16LittleEndian(response);
+            if ((i & (1 << 15)) == (1 << 15))
+            {
+                i -= (1 << 16);
+            }
+
+            return new ElectricCurrent(i, ElectricCurrentUnit.Milliampere);
+        }
+
+        /// <summary>
+        /// Get the color for a specific Led
+        /// </summary>
+        /// <param name="led">Led to get color for</param>
+        /// <returns>Color of Led</returns>
+        public Color GetLedState(Led led)
+        {
+            var response = _piJuice.ReadCommand(PiJuiceCommand.LedState + (byte)led, 3);
+
+            return Color.FromArgb(0, response[0], response[1], response[2]);
+        }
+
+        /// <summary>
+        /// Set the color for a specific Led
+        /// </summary>
+        /// <param name="led">Led to which color is to be applied</param>
+        /// <param name="color">Color for the Led</param>
+        public void SetLedState(Led led, Color color)
+        {
+            _piJuice.WriteCommand(PiJuiceCommand.LedState + (byte)led, new byte[] { color.R, color.G, color.B });
+        }
+
+        /// <summary>
+        /// Get blinking pattern for a specific Led
+        /// </summary>
+        /// <param name="led">Led to get blinking pattern for</param>
+        /// <returns>Led blinking pattern</returns>
+        public LedBlink GetLedBlink(Led led)
+        {
+            var response = _piJuice.ReadCommand(PiJuiceCommand.LedBlink + (byte)led, 9);
+
+            return new LedBlink
+            {
+                Led = led,
+                BlinkIndefinite = response[0] == 255,
+                Count = response[0],
+                ColorFirstPeriod = Color.FromArgb(0, response[1], response[2], response[3]),
+                FirstPeriod = new TimeSpan(0, 0, 0, 0, response[4] * 10),
+                ColorSecondPeriod = Color.FromArgb(0, response[5], response[6], response[7]),
+                SecondPeriod = new TimeSpan(0, 0, 0, 0, response[8] * 10)
+            };
+        }
+
+        /// <summary>
+        /// Set blinking pattern for a specific Led
+        /// </summary>
+        /// <param name="ledBlink">Led blinking pattern</param>
+        public void SetLedBlink(LedBlink ledBlink)
+        {
+            if (ledBlink.Count < 1 || ledBlink.Count > 255)
+            {
+                throw new ArgumentOutOfRangeException(nameof(ledBlink.Count));
+            }
+
+            if (ledBlink.FirstPeriod.TotalMilliseconds < 10 || ledBlink.FirstPeriod.TotalMilliseconds > 2550)
+            {
+                throw new ArgumentOutOfRangeException(nameof(ledBlink.FirstPeriod));
+            }
+
+            if (ledBlink.SecondPeriod.TotalMilliseconds < 10 || ledBlink.SecondPeriod.TotalMilliseconds > 2550)
+            {
+                throw new ArgumentOutOfRangeException(nameof(ledBlink.SecondPeriod));
+            }
+
+            var data = new byte[9];
+
+            data[0] = (byte)(ledBlink.Count & 0xFF);
```

Comment:
should you throw if this is higher than 255?

Reply:
you do throw actually, if so then why not just cast since it can't overflow?

Reply:
I think it would make more sense to change Count from a short to a byte